modules/average_month.py

```diff
@@ -0,0 +1,93 @@
+"""
+This module generates a report which displays average highest and lowest temperature
+as well as average mean humidity for a given month
+"""
+from constants import WEATHER_FILES_DIR, months_list
+from modules.utils import (
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_mean_humidity,
+    read_data,
+)
+
+
+def calculate_averages(pattern, path):
+    """
+    Calculates average of highest_temperature, lowest_temperature and mean_humidity
+    from files with names that match a pattern in given path
+    1. avg highest temperature
+    2. avg lowest temperature
+    3. avg mean humidity
+    Args:
+        pattern(str): a string which contains a year/month e.g '2006/5', '2007/3'
```

Comment:
Let's make it even cleaner, Pass year and month as 2 separate params. Also, it would be a good call to check the validation on month and year e.g. either they were parsed or not -- This validation could be done in the caller function or through a generic utils method.